src/senders/report_stream_sender_test.go

```diff
@@ -413,8 +413,146 @@ func (suite *SenderTestSuite) Test_SendMessage_StatusCodeIsAbove300_ReturnsError
 
 	reportId, err := sender.SendMessage(message)
 
-	assert.Equal(suite.T(), "404 Not Found", err.Error())
 	assert.Error(suite.T(), err)
+	assert.Equal(suite.T(), "302 Found", err.Error())
+	assert.Equal(suite.T(), "", reportId)
+}
+
+func (suite *SenderTestSuite) Test_SendMessage_StatusCodeIs400_ReturnsNonTransientError() {
+	sender, err := NewSender()
+	assert.NoError(suite.T(), err)
```

Comment:
Is it necessary to check for errors after each step if it's not the target of the test?

Reply:
Asserts on test setup have been removed